src/hooks/use-cart.js

```diff
@@ -36,21 +36,39 @@ export const useCart = (user = null) => {
     const { discount, categories } = promoCode.getPromoCodeByCode;
     const newArr = cart.map((item) => {
       const { price } = item.sizeAndPrice;
-
+      
       const isAllowCategory = categories.find(
         (el) => el.toLowerCase() === item.category.code.toLowerCase()
       );
       if (isAllowCategory) {
-        return [Math.round(price - (price / 100) * discount), item.quantity];
+        return [getPriceWithCurrency(Math.round(price - (price / 100) * discount)), item.quantity];
+        
       }
-      return [Math.round(price), item.quantity];
+      return [getPriceWithCurrency(price), item.quantity];
     });
     return newArr.reduce(
-      (acc, item) => acc + getPriceWithCurrency(calcPriceForCart(item[0], item[1])),
+      (acc, item) => {
+         return acc + calcPriceForCart(item[0], item[1])},
```

Comment:
I don't think its intuitively clear what item[0],..etc mean. Maybe it make sence to destructure them first?